### ISSUE TITLE
Decision Tree Test Values that weren't in Training Set

### ISSUE THREAD

**Issue by drusk** [drusk/pml#58]
A KeyError is being thrown if a feature value is encountered that wasn't in the training set.  Need to find out the standard strategy for dealing with this.

**Comment by drusk** [comment]
http://webcache.googleusercontent.com/search?q=cache:nUKuzJi9-0UJ:www.cs.colorado.edu/~mozer/Teaching/syllabi/5622/assign5.ps+&cd=1&hl=en&ct=clnk&gl=ca

says:
"You may need to handle the case where decisions are based on an attribute dimension which has val-
ues in the test set that weren’t contained in the training set. In this case, you should classify the exam-
ple according to the majority value in the root node."

This will be a simple solution for now.  Will need to keep an eye out for other solutions.  I wonder if it would be better to take the majority vote of training examples which would qualify to reach the node at which the problem occurs.

**Comment by drusk** [comment]
This solution does not seem ideal but will do for the moment.  This should be investigated further at some point time permitting.
